ClientRuntimes/NodeJS/ms-rest-azure/lib/azureServiceClient.js

```diff
@@ -233,36 +252,25 @@ AzureServiceClient.prototype._updateStateFromAzureAsyncOperationHeader = functio
 };
 
 /**
- * Retrieve PUT operation status by polling from 'location' header.
+ * Retrieve PUT/POST/PATCH/DELETE operation status by polling from 'location' header.
  * @param {object} [pollingState] - The object to persist current operation state.
  */
-AzureServiceClient.prototype._updateStateFromLocationHeaderOnPut = function (pollingState, callback) {
-  this._getStatus(pollingState.locationHeaderLink, function (err, result) {
+AzureServiceClient.prototype._updateStateFromLocationHeader = function(pollingState, callback) {
+  this._getStatus(pollingState.locationHeaderLink, function(err, result) {
     if (err) return callback(err);
-    
+
     pollingState.updateResponse(result.response);
     pollingState.request = result.request;
-    
+
     var statusCode = result.response.statusCode;
     if (statusCode === 202) {
       pollingState.status = LroStates.InProgress;
-    }
-    else if (statusCode === 200 ||
-             statusCode === 201) {
-      
-      if (!result.body) {
-        return callback(new Error('The response from long running operation does not contain a body.'));
-      }
-      
-      // In 202 pattern on PUT ProvisioningState may not be present in 
-      // the response. In that case the assumption is the status is Succeeded.
-      if (result.body.properties && result.body.properties.provisioningState) {
-        pollingState.status = result.body.properties.provisioningState;
-      }
-      else {
-        pollingState.status = LroStates.Succeeded;
-      }
-      
+    } else if (statusCode === 200 ||
+      statusCode === 201 ||
+      statusCode === 204) {
```

Comment:
204 is an error case for put, sine you are merging "put" and "delete" logic into one method, then this is behavior change.